app/pages/Page.scala

```diff
@@ -16,12 +16,18 @@
 
 package pages
 
-import models.Mode
+import models.backend.RetrievedSubscription
+import models.{Mode, UserAnswers}
+import queries.Query
 
 import scala.language.implicitConversions
 
+case class RequiredPage(page: Page with Query, additionalPreconditions: List[Boolean] = List.empty)
+
 trait Page {
   val url: Mode => String = mode => controllers.routes.IndexController.onPageLoad.url
+
+  val previousPagesRequired: (UserAnswers, RetrievedSubscription) => List[RequiredPage] = (_, _) => List.empty
 }
```

Comment:
Is it possible to put the val stuff for amount required?  If journey == changeActivity ...
  override val previousPagesRequired: (UserAnswers, RetrievedSubscription) => List[RequiredPage] = (userAnswers, _) => {
    val isSmallOrLargeProducer: Boolean =
      userAnswers.get(AmountProducedPage).contains(AmountProduced.Large) || userAnswers.get(AmountProducedPage).contains(AmountProduced.Small)
    val isSmallOrNoneProducer: Boolean =
      userAnswers.get(AmountProducedPage).contains(AmountProduced.Small) || userAnswers.get(AmountProducedPage).contains(AmountProduced.None)
    }).getOrElse(false)

Reply:
If it is too much to implement or causes problems for the other journeys, no worries, it would just help reduce duplication if it is possible